File: applications/UserManagement/Authorization.Common/Models/ApplicationModel.cs
namespace Authorization.Common.Models;

/// <summary>
/// DTO Class for Application Entity
/// </summary>
public class ApplicationModel
{
    /// <summary>
    /// Unique Identifier for the Application Model.
    /// </summary>
    public Guid Id { get; set; }

    /// <summary>
    /// Name of the Application.
    /// </summary>
    public string Name { get; set; } = null!;

    /// <summary>
    /// Application Description.
    /// </summary>
    public string Description { get; set; } = default!;

    /// <summary>
    /// Defines whether the application should support Client Credential Auth (Service-to-Service Authentication).
    /// </summary>
    public bool SupportClientAuthentication { get; set; }
}

File: applications/UserManagement/Authorization.TwinPlatform.Web/authorization-client-app/src/types/ApplicationModel.ts
import { Guid } from "./Guid";

export class ApplicationModel {
  id: string = Guid.Empty.ToString();
  name: string = '';
  description: string = '';
  supportClientAuthentication: boolean = false;

  static MapModel(apiRow: any): ApplicationModel {
    let newRow = new ApplicationModel();
    Object.assign(newRow, apiRow);
    return newRow;
  }
}
